python/lsst/ts/externalscripts/auxtel/correct_pointing.py

```diff
@@ -0,0 +1,290 @@
+__all__ = [
+    "CorrectPointing",
+]
+
+import asyncio
+import warnings
+
+import numpy as np
+from lsst.geom import PointD
+
+try:
+    from lsst.pipe.tasks.quickFrameMeasurement import QuickFrameMeasurementTask
+    from lsst.summit.utils import BestEffortIsr
+    from lsst.ts.observing.utilities.auxtel.latiss.getters import get_image
+    from lsst.ts.observing.utilities.auxtel.latiss.utils import (
+        calculate_xy_offsets,
+        parse_visit_id,
+    )
+except ImportError:
+    warnings.warn("Cannot import required libraries. Script will not work.")
+
+
+from lsst.ts.observatory.control.auxtel import ATCS, LATISS, ATCSUsages, LATISSUsages
+from lsst.ts.observatory.control.constants.latiss_constants import boresight
+from lsst.ts.observatory.control.utils.enums import RotType
+from lsst.ts.salobj import BaseScript
+
+
+class CorrectPointing(BaseScript):
+    """Find the zero point offset of the pointing at the start of the night."""
+
+    def __init__(self, index: int, remotes: bool = True) -> None:
+
+        super().__init__(
+            index=index,
+            descr="Correct pointing.",
+        )
+
+        if remotes:
+            self.atcs = ATCS(domain=self.domain, log=self.log)
+            self.latiss = LATISS(domain=self.domain, log=self.log)
+        else:
+            self.atcs = ATCS(
+                domain=self.domain, log=self.log, intended_usage=ATCSUsages.DryTest
+            )
+            self.latiss = LATISS(
+                domain=self.domain, log=self.log, intended_usage=LATISSUsages.DryTest
+            )
+
+        self.image_in_oods_timeout = 15.0
+        self.get_image_timeout = 10.0
+        self.tolerance = 1.0
+        self.exposure_time = 1.0
+
+        self.magnitude_limit = 8
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    def set_metadata(self, metadata):
+        pass
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        self.best_effort_isr = self.get_best_effort_isr()
+
+    def get_best_effort_isr(self):
+        # Isolate the BestEffortIsr class so it can be mocked
+        # in unit tests
+        return BestEffortIsr()
+
+    @property
+    def camera_readout_time(self):
+        return 2.0
+
+    async def arun(self, checkpoint_active=False):
+        """Execute script main operations.
+
+        Parameters
+        ----------
+        checkpoint_active : bool, optional
+            Run script with checkpoints, by default False?
+        """
+
+        await self.handle_checkpoint(
+            checkpoint_active=checkpoint_active,
+            checkpoint_message="Setting up.",
+        )
+
+        await asyncio.gather(
+            self._setup_atcs(),
+            self._setup_latiss(),
+        )
+
+        await self.handle_checkpoint(checkpoint_active, "Correcting pointing...")
```

Comment:
I think the current message is more appropriate for a checkpoint.. because one can pause here by requesting a pause in "Correcting *", which I think makes more sense than in "Adding *"....

Reply:
well, it is the last point to stop before the Script jumps into the correction loop... it's kind of the last place to pause the execution..